support/primitives/src/lib.rs

```diff
@@ -12,17 +11,17 @@ mod tests;
 
 /// A struct that implements the `CommitmentVerifier` trait, where the commitment
 /// is a Merkle Patricia Trie root hash.
-pub struct TrieVerifier<T: TrieLayout>
+pub struct TrieVerifier<T: TrieLayout, const H_LENGTH: usize>
 where
-    <T::Hash as sp_core::Hasher>::Out: TryFrom<Vec<u8>>,
+    <T::Hash as sp_core::Hasher>::Out: for<'a> TryFrom<&'a [u8; H_LENGTH]>,
 {
     pub _phantom: core::marker::PhantomData<T>,
 }
 
 /// Implement the `CommitmentVerifier` trait for the `TrieVerifier` struct.
-impl<T: TrieLayout> CommitmentVerifier for TrieVerifier<T>
+impl<T: TrieLayout, const H_LENGTH: usize> CommitmentVerifier for TrieVerifier<T, H_LENGTH>
 where
-    <T::Hash as sp_core::Hasher>::Out: TryFrom<Vec<u8>>,
+    <T::Hash as sp_core::Hasher>::Out: for<'a> TryFrom<&'a [u8; H_LENGTH]>,
```

Comment:
What is the reason behind this?

Reply:
If I remember correctly, it is because `H256` doesn't implement `TryFrom<Vec<u8>>`, but it implements `TryFrom<[u8; 32]>`